### PR TITLE
Wrap combo reduct

### PR DESCRIPTION
**Pull request by kasimebrahim**
Some reductions are not ported yet, because the converter doesn't support certain vertex types and it is impossible to test with out it.

### CODE REVIEW

File: tests/reduct/ContinReductUTest.cxxtest [hunk 1]
```diff
@@ -31,209 +31,293 @@
...
+
+	void perform_test_reduce(Handle &h, bool strict = true)
+	{
+		int size_before, size_after;
+		int reduct_effort = 2;
+
+		AtomeseToCombo to_combo;
+		combo_tree tr_before = to_combo(h).first;
+
+		type_tree tt = infer_type_tree(tr_before);
```
Comment (ngeiswei) [inline tests/reduct/ContinReductUTest.cxxtest]: That makes me think we need some atomese equivalent of `infer_type_tree` we already have some type inference code in atomspace but I don't think it is good enough. Anyway, that's a concern for later.

File: tests/reduct/ContinReductUTest.cxxtest [hunk 2]
```diff
@@ -31,209 +31,293 @@
...
+		ITable rnt(tt);
+
+		contin_reduce(h, reduct_effort, HandleSet());
+
+		combo_tree tr_after = to_combo(h).first;
+
+		try {
+			OTable ct1(tr_before, rnt);
```
Comment (ngeiswei) [inline tests/reduct/ContinReductUTest.cxxtest]: And we probably want to overload OTable to support atomese, but again, let's worry about later.

File: opencog/reduct/reduct/contin_reduction.cc [hunk 3]
```diff
@@ -31,107 +31,127 @@
...
-namespace opencog { namespace reduct {
+namespace opencog
+{
+namespace reduct
+{
+
+const rule &contin_reduction(int reduct_effort,
+                             const vertex_set &ignore_ops)
+{
+	return setup_contine_reduction(reduct_effort, ignore_ops);
```
Comment (ngeiswei) [inline opencog/reduct/reduct/contin_reduction.cc]: There's a typo here `setup_contine_reduction` -> `setup_contin_reduction`. I know you didn't introduce it but could I ask you to fix it?

**Comment by ngeiswei** [comment]
BTW, any idea which the unit test fails?

**Comment by kasimebrahim** [comment]
It is https://github.com/singnet/as-moses/blob/fb9d440a6628da3e48e08638d1ca172950e661f6/tests/moses/main/ASMOSESUTest.cxxtest#L100 
The problem is the converter doesn't support some vertices yet i:e id::exp. we will have to implement them to fix it.

**Comment by ngeiswei** [comment]
OK, I remember now, thanks.

File: opencog/reduct/reduct/reduct.h [hunk 4]
```diff
@@ -109,7 +110,11 @@ struct logical_reduction
 
 	logical_reduction(const vertex_set &ignore_ops);
 
-	logical_reduction(const HandleSet &ignore_ops);
+	logical_reduction(const HandleSet &ignore_ops)
+			: logical_reduction(set_comp(ignore_ops, [this](const Handle h) {
+		return *to_combo(h).first.begin();
+	}))
+	{}
```
Comment (ngeiswei) [inline opencog/reduct/reduct/reduct.h]: Really minor point but it would probably have been better to have this code in `logical_reduction.cc`.

**Pull request merged**